src/System.Net.Http/tests/StressTests/HttpStress/Program.cs

```diff
@@ -0,0 +1,377 @@
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using System;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Security;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Diagnostics.Tracing;
+using System.Text;
+using System.Net;
+
+/// <summary>
+/// Simple HttpClient stress app that launches Kestrel in-proc and runs many concurrent requests of varying types against it.
+/// Usage:
+///     dotnet run -c Release [-t [path]]
+/// e.g.
+///     dotnet run -c Release
+///     dotnet run -c Release -t
+///     dotnet run -c Release -t d:\log.txt
+/// </summary>
+public class Program
+{
+    public static void Main(string[] args)
+    {
+        // Print out environment information
+        Console.WriteLine("   .NET Core: " + Path.GetFileName(Path.GetDirectoryName(typeof(object).Assembly.Location)));
+        Console.WriteLine("ASP.NET Core: " + Path.GetFileName(Path.GetDirectoryName(typeof(WebHost).Assembly.Location)));
+
+        // Handle command-line arguments.
+        EventListener listener = null;
+        if (args.Length > 0)
+        {
+            switch (args[0])
+            {
+                case "-t":
+                    listener = new HttpEventListener(args.Length > 1 ? new StreamWriter(args[1]) { AutoFlush = true } : null);
+                    break;
+            }
+        }
+
+        // Basic configuration
+        const int ConcurrentRequests = 150;
+        const int AppxContentLength = 1_000;
+        var httpVersion = new Version(2, 0);
+        string content = string.Concat(Enumerable.Repeat("1234567890", AppxContentLength / 10));
+        const int DisplayIntervalMilliseconds = 1000;
+
+        const int HttpsPort = 5001;
+        const string LocalhostName = "localhost";
+        string serverUri = $"https://{LocalhostName}:{HttpsPort}";
+
+        // Validation of a response message
+        void ValidateResponse(HttpResponseMessage m)
+        {
+            if (m.Version != httpVersion)
+            {
+                throw new Exception($"Expected response version {httpVersion}, got {m.Version}");
+            }
+        }
+
+        // Set of operations that the client can select from to run.  Each item is a tuple of the operation's name
+        // and the delegate to invoke for it, provided with the HttpClient instance on which to make the call and
+        // returning asynchronously the retrieved response string from the server.  Individual operations can be
+        // commented out from here to turn them off, or additional ones can be added.
+        var clientOperations = new (string, Func<HttpClient, Task<string>>)[]
+        {
+            ("GET",
+            client => client.GetStringAsync(serverUri)),
```

Comment:
nit: any reason why this doesn't call ValidateResponse like the other cases?

Reply:
This API doesn't give me access to the HttpResponseMessage. I could switch to not using the helper, but then we're not exercising this common method. We could also add another case.

Reply:
Meh, I'll just such to using the underlying methods.